bpf-gpl/conntrack.h

```diff
@@ -243,6 +284,9 @@ static CALI_BPF_INLINE int calico_ct_v4_create_nat_fwd(struct ct_create_ctx *ct_
 
 	dump_ct_key(&k);
 	ct_value.nat_rev_key = *rk;
+	if (ct_ctx->orig_sport) {
```

Comment:
Yeah, the intention of this if is to set the entry so that it keeps changing the port  for subsequent packets only if there was a collision. It is meaningless as is, since it is always set (great catch) and it sets it for all entries. I will fix that.